### PR TITLE
upgrade nodejs deprecations

### PR DESCRIPTION
**Pull request by stevie-**
hi, here is our take of your action.

you may want to adapt some things in your action.

### CODE REVIEW

**Comment by GuillaumeFalourd** [comment]
Hi @stevie- , thank you for contributing to the `pull-request-action` repository!

Updates related to the `action.yaml` file looks good to me. However, I wouldn't update the 3 workflows to test the action in each runner, neither the `README.md` file.

Would you mind If I just update the action using the `action.yaml` file implementation you suggested? 😃

File: action.yml [hunk 1]
```diff
@@ -38,34 +38,47 @@ inputs:
...
+        echo "::group::Gather Inputs"
+        if [[ ! -z "$GITHUB_HEAD_REF" ]]; then
+          SOURCE_BRANCH="$GITHUB_HEAD_REF"
+        elif [[ ! -z "$INPUT_SOURCE_BRANCH" ]]; then
+          SOURCE_BRANCH="$INPUT_SOURCE_BRANCH"
         elif [[ ! -z "$GITHUB_REF" ]]; then
           SOURCE_BRANCH=${GITHUB_REF/refs\/heads\//}  # Remove branch prefix
         else
-          echo "Set the ${{ inputs.source_branch }} environment variable or trigger from a branch."
+          echo_fail "Set the INPUT_SOURCE_BRANCH environment variable or trigger from a branch."
```
Comment (stevie-) [inline action.yml:69]:
```suggestion
          echo "Set the INPUT_SOURCE_BRANCH environment variable or trigger from a branch."
```

**Comment by stevie-** [comment]
Here is another version which is more close to the original code
https://github.com/datadrivers/pull-request-action/compare/v2.0.1...v2.1.0

**Comment by GuillaumeFalourd** [comment]
> Here is another version which is more close to the original code including bug fixes like handling multi line inputs https://github.com/datadrivers/pull-request-action/releases/tag/v2.1.3

Thanks! I've created a [test](https://github.com/GuillaumeFalourd/pull-request-action/tree/test) branch using exactly the [action.yaml file](https://github.com/GuillaumeFalourd/pull-request-action/blob/test/action.yml) implementation you suggested.

If you prefer updating this PR to push just this `action.yml` file to the repository, you'll appear as contributor 👍🏼

**Comment by stevie-** [comment]
your branch looks good

**Pull request closed without merging**